server/src/controller/comment.ts

```diff
@@ -0,0 +1,85 @@
+import { QueryRunner } from "typeorm";
+import { ErrorType, SuccessType, CommentCreate } from "../types";
+import AppDataSource from "../db";
+import { Comment } from "../model";
+import { getOneCustomer } from "./customer";
+import { getOneItem } from "./item";
+
+const queryRunner: QueryRunner = AppDataSource.createQueryRunner();
+const commentRepository = AppDataSource.getRepository(Comment);
+
+export async function createComment(
+  body: CommentCreate,
+): Promise<ErrorType | SuccessType> {
+  try {
+    await queryRunner.startTransaction();
+    // eslint-disable-next-line no-console
+    console.log(body);
+
+    const comment = new Comment();
+
+    comment.comments = body.comments;
+
+    if (body.customerId) {
+      const customerId = await getOneCustomer(body.customerId);
+
+      if (customerId === null)
+        return {
+          errorCode: 404,
+          errorDescription: "Customer not found.",
+          errorKey: "Customer",
+        };
+      comment.customer = customerId;
+    }
+
+    if (body.itemId) {
+      const itemId = await getOneItem(body.itemId);
+
+      if (itemId === null)
+        return {
+          errorCode: 404,
+          errorDescription: "Item not found.",
+          errorKey: "Item",
+        };
+      comment.item = itemId;
+    }
+
+    await queryRunner.manager.save(comment);
+
+    await queryRunner.commitTransaction();
+    return {
+      data: {
+        id: comment.id,
+        comments: comment.comments,
+        customerId: comment.customer ? comment.customer : null,
+        itemId: comment.item ? comment.item : null,
+        createdAt: comment.created_at,
+      },
+    };
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    await queryRunner.rollbackTransaction();
+    if (error.code === "23505") {
```

Comment:
here is where you know there is a duplicate comment, just add the unique decorator at the top of the comments model